bottomless/src/bottomless_wal.rs

```diff
@@ -80,16 +82,19 @@ impl<T: Wal> WrapWal<T> for BottomlessWalWrapper {
         Ok(())
     }
 
-    fn checkpoint<B: libsql_sys::wal::BusyHandler>(
+    fn checkpoint(
         &mut self,
         wrapped: &mut T,
-        db: &mut libsql_sys::wal::Sqlite3Db,
-        mode: libsql_sys::wal::CheckpointMode,
-        busy_handler: Option<&mut B>,
+        db: &mut Sqlite3Db,
+        mode: CheckpointMode,
+        busy_handler: Option<&mut dyn BusyHandler>,
```

Comment:
Is `dyn BusyHandler` required by this series, or just a piggy-back since checkpoint callback is getting dynamic dispatch? If it's unrelated then I'll just ignore it in the first review pass

Reply:
no, it is actually required. Not using it means that there would potentially be multiple implementation of the checkpoint method, which is not correct form the perspective of the virtual methods